### PR TITLE
TEST-#6030: test the function parameters of `Series.str` accessor for pandas equivalence

### LINKED ISSUE #6030: Test the function parameters of `Series.str` accessor for pandas equivalence

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

I guess I'll split this PR into 3 parts:
* `Series.str` implementation through `cached_accessor`
* signatures correction
* adding tests

However, I think it will be more convenient to review the changes together.

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6030 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/series_utils.py [hunk 1]
```diff
@@ -40,6 +39,7 @@
 class CategoryMethods(ClassLogger):
     def __init__(self, series):
         self._series = series
+        self._series_type = type(series)
```
Comment (anmyachev) [inline modin/pandas/series_utils.py:42]: This change is necessary to avoid cyclic imports.

File: modin/pandas/series_utils.py [hunk 2]
```diff
@@ -485,115 +516,122 @@ def _default_to_pandas(self, op, *args, **kwargs):
 class DatetimeProperties(ClassLogger):
     def __init__(self, series):
         self._series = series
+        self._series_type = type(series)
         self._query_compiler = series._query_compiler
 
     @property
     def date(self):
-        return Series(query_compiler=self._query_compiler.dt_date())
+        return self._series_type(query_compiler=self._query_compiler.dt_date())
```
Comment (dchigarev) [inline modin/pandas/series_utils.py:524]: maybe like this?
```suggestion
        return self._series.__constructor__(query_compiler=self._query_compiler.dt_date())
```
it seems to be the pattern we're using everywhere in the probject

File: modin/pandas/test/test_api.py [hunk 3]
```diff
@@ -51,6 +51,7 @@ def test_top_level_api_equality():
         "iterator",
         "series",
         "accessor",
+        "series_utils",
```
Comment (dchigarev) [inline modin/pandas/test/test_api.py:54]: so how it used to work before?

**Review by dchigarev — APPROVED** [review]
lgtm!

left two minor comments

**Comment by anmyachev** [comment]
Ok, I'm starting to split this pull request into parts.

**Comment by anmyachev** [comment]
@dchigarev ready for review

**Review by dchigarev — APPROVED** [review]

**Pull request merged**